__test__/pet.test.js

```diff
@@ -0,0 +1,132 @@
+const Pet = require ('../src/pet.js')
+
+describe('constructor', () => {
+    it('returns an object', () => {
+        expect(new Pet ('Bob')).toBeInstanceOf(Object);
+    });
+    it('sets the name property', () => {
+        const pet = new Pet('Bob');
+        expect(pet.name).toEqual('Bob');
+      });
+    it('has an initial age of 0', () => {
+      const pet = new Pet('Bob')
+      pet.growUp()
+      expect(pet.age).toEqual(1);
+    })
+    it('has inital hunger of 0', () => {
+      const pet = new Pet ('Bob')
+      pet.getHungry()
+      expect(pet.hunger).toEqual(5)
+    })
+    it('has inital fitness of 10', () => {
+      const pet = new Pet ('Bob')
+      pet.becomeUnfit()
+      expect(pet.fitness).toEqual(7)
+    })
+    });
+
+    describe('walk', () => {
+      it('increases fitness by 4', () => {
+        const pet = new Pet('Bob')
+        pet.fitness = 4;
+        pet.walk()
+        expect(pet.fitness).toEqual(8)
+      })
+      it('it increases fitness to a maximum of 10', () => {
+        const pet = new Pet ('Bob')
+        pet.fitness = 8;
+        pet.walk()
+        expect(pet.fitness).toEqual(10)
+      })
+      it('it throws an error if the pet is not alive', () => {
+        const pet = new Pet ('Bob')
+        pet.age = 30
+        expect(() => pet.walk()).toThrow('Your pet is no longer alive')
+      })
+    })
+
+    describe('grow up', () => {
+      it('increases the age of the pet in incrememnts of 1', () => {
+        const pet = new Pet('Bob');
+        pet.growUp();
+        expect(pet.age).toEqual(1);
+      });
+      it('it throws an error if pet is no longer alive', () => {
+        const pet = new Pet('Bob');
+        pet.age = 30;
+        expect(() => pet.growUp()).toThrow('Your pet is no longer alive');
+      });
+    });
+
+    describe('feed', () => {
+      it('it prevents hunger level from going below 0', () => {
+        const pet = new Pet ('Bob')
+        pet.hunger = 8
+        pet.feed()
+        expect(pet.hunger).toEqual(0)
+      })
+      it('it throws an error if the pet is not alive', () => {
+        const pet = new Pet('Bob')
+        pet.age = 30
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive')
+      })
+    })
+
+    describe('checkUp', () => {
+      it('returns I need a walk if fitness is less than 3', () => {
+        const pet = new Pet ('Bob')
+        pet.fitness = 8
+        pet.checkUp()
+        expect('I need a walk!')
+      })
```

Comment:
This test name doest make sense, as youre setting the fitness level to 8 and asserting based on that. Also, this is passing by accident as you arent checking what the checkUp function is returning. To do that, you should assert this:
```
    expect(pet.checkUp()).toEqual("I need a walk!");
```
Putting that into the function, you get this:
```
    it("returns I need a walk if fitness is less than 3", () => {
        const pet = new Pet("Bob");
        pet.fitness = 8;

        expect(pet.checkUp()).toEqual("I need a walk!");
    });
```
This test will fail when you update to this as the fitness level of 8 shouldnt return this message. If it is set to 3, it will pass as expected